util/alertifier.go

```diff
@@ -0,0 +1,138 @@
+package util
+
+// DCSO FEVER
+// Copyright (c) 2020, DCSO GmbH
+
+import (
+	"fmt"
+
+	"github.com/DCSO/fever/types"
+
+	"github.com/buger/jsonparser"
+	log "github.com/sirupsen/logrus"
+)
+
+// ExtraModifier is a function type that describes a function that adds the
+// appropriate `_extra` sub-object entries to a EVE-JSON event.
+type ExtraModifier func(inputAlert *types.Entry, ioc string) error
+
+// AlertJSONProvider is an interface describing a component that returns an
+// `alert` JSON sub-object to use in an EVE-JSON event.
+type AlertJSONProvider interface {
+	// GetAlertJSON is a function that returns a byte slice containing the
+	// JSON data for an `alert` EVE-JSON sub-object.
+	GetAlertJSON(inputEvent types.Entry, prefix string, ioc string) ([]byte, error)
+}
+
+// Alertifier is a component that creates EVE-JSON alerts from arbitrary
+// EVE-JSON events. It does this by cloning the original event and adding
+// alert-specific fields, depending on the given ExtraModifier and a set of
+// AlertJSONProviders, selectable using a string tag.
+type Alertifier struct {
```

Comment:
In Go, types ending with `-er` are _typically_ Interface types. Maybe `EventAlert`?

Bit late to the game, is not so importan. I saw `*util.Alertifier` and that did not compute with me brain that quickly.

Reply:
> In Go, types ending with `-er` are typically Interface types. Maybe `EventAlert`?

Huh? Where is this convention written down? Didn't know about that and I haven't seen that in any other Go code I read.
I used the `-er` name here to signify that this is an active component that creates alerts from events, and does not represent an alert itself (alerts are just `types.Entry` instances with a specific `event_type` set). That's all.

Reply:
Not so important, but here it is https://golang.org/doc/effective_go.html#interface-names